config/main.py

```diff
@@ -1642,12 +1642,54 @@ def breakout(ctx, interface_name, mode, verbose, force_remove_dependencies, load
 
     """ Load config for the commands which are capable of change in config DB """
     cm = load_configMgmt(verbose)
+
+    """ Save Port OIDs Mapping Before Deleting Port"""
+    dataBase = SonicV2Connector(host="127.0.0.1")
+    if_name_map, if_oid_map = port_util.get_interface_oid_map(dataBase)
+
     """ Delete all ports if forced else print dependencies using configMgmt API """
     final_delPorts = [intf for intf in del_intf_dict.keys()]
     delete_Ports(cm, final_delPorts, force_remove_dependencies, verbose)
 
-    # TODOFIX:  Check ASIC DB whether interfaces got deleted or not
-    time.sleep(5)
+    """
+     Internal Function:  Check ASIC DB whether interfaces present or not
+     ports: List of ports
+     presence: If False, Make sure Ports are not present, return False otherwise.
+               If True, Make sure Ports are present, return True otherwise.
+    """
+    def check_Ports_AsicDb(db, ports, presence, portMap):
+
+        try:
+            db.connect(db.ASIC_DB)
+            oidKey = 'ASIC_STATE:SAI_OBJECT_TYPE_PORT:oid:0x*'
+            oidDict = {oids.strip(oidKey):True for oids in \
+                db.keys('ASIC_DB', oidKey)}
+
+            for port in ports:
+                oid = portMap.get(port)
+                print("For Testing [Check Asic DB] {}:{}".format(port, oid))
+                # If Presence == False but entry exists, return False
+                if presence == False and oidDict.get(oid):
+                    return False
+                # If Presence == True and entry does not exists, return False
+                elif presence and oidDict.get(oid) == None:
```

Comment:
when would the value of `presence` be true? As per `check_Ports_AsicDb` function, the default value of `presence` is `False` which never gets True by code. Am I missing something?

Reply:
Same API can be be used to make sure that ports are present in ASIC DB. Right now we do not call it with True because we do not check PORT after ADD API.

Reply:
Why cant this function be as simple as, return true if port exists otherwise false. Lets the user use it the way he/she wants it!

Reply:
Caller has to write 'for loop' each time. To avoid that I added for loop inside function. And once for loop is inside, we need presence flag.